NDTensors/src/default_storage.jl

```diff
@@ -0,0 +1,19 @@
+## This is a fil which specifies the default storage type provided some set of parameters
+## The parameters are the element type and storage type
+default_datatype(eltype::Type{<:Number}) = Vector{eltype}
+default_eltype() = Float64
+
+default_storagetype(datatype::Type{<:AbstractArray{ElT}}) where {ElT} = Dense{ElT, datatype}
+
+default_storagetype(datatype::Type{<:AbstractArray}) = default_storagetype(datatype{default_eltype()})
+default_storagetype(eltype::Type{<:Number}) = default_storage_type(default_datatype(eltype))
+default_storagetype() = default_storagetype(default_datatype(default_eltype()))
+default_storage_type(storagetype::Type{<:TensorStorage}) = 
```

Comment:
This function may be beyond the type of functionality we want for the `default_storage_type`. I think `default_storage_type` should only handle hierarchies, not converting/adapting storage types.

Instead, I think this functionality could be handled by a function `adapt_type` (which would supersede the current `NDTensors.similartype` function):
```julia
adapt_type(default_datatype(), storagetype)
```

Reply:
Open to discussion about it though. A lot of this design depends on how it will actually get used in the constructors, so I think it will be helpful to see how that works out.

Reply:
I think that this functionality would be useful, but at the moment complicated to implement in this matter. I saw your comments the day before and didn't modify this `default_storage_type` function from the previous push. For now I will remove it and I think it makes sense to utilize an `adapt_type` function to perform this operation.